src/QuiqChatClient.js

```diff
@@ -215,25 +232,46 @@ class QuiqChatClient {
   };
 
   _handleConnectionLoss = () => {
+    this.connected = false;
+
     if (this.callbacks.onConnectionStatusChange) {
       this.callbacks.onConnectionStatusChange(false);
     }
   };
 
-  _handleConnectionEstablish = () => {
-    const messages = getConversationMessages();
+  _handleConnectionEstablish = async () => {
+    const {messages, events} = await getConversation();
 
-    // If messages came in while disconnected, push to callback
-    const newMessages = differenceBy(this.messages, messages, 'id');
+    this._processNewMessagesAndEvents(messages, events);
 
-    if (newMessages.length && this.callbacks.onNewMessages) {
-      this.callbacks.onNewMessages(newMessages);
-    }
+    this.connected = true;
 
     if (this.callbacks.onConnectionStatusChange) {
       this.callbacks.onConnectionStatusChange(true);
     }
   };
+
+  _processNewMessagesAndEvents = (
+    messages: Array<TextMessage>,
+    events: Array<Event> = [],
+    sendNewMessageCallback: boolean = true,
+  ): void => {
+    const newMessages = differenceBy(messages, this.textMessages, 'id');
+    const newEvents = differenceBy(events, this.events, 'id');
+
+    if (newMessages.length && this.callbacks.onNewMessages && sendNewMessageCallback) {
+      this.callbacks.onNewMessages(newMessages);
```

Comment:
do the newMessages need to be sorted by timestamp like they are below after the callback is called?